cocos/asset/assets/image-asset.ts

```diff
@@ -609,56 +617,58 @@ export class ImageAsset extends Asset {
 
     private _height = 0;
 
-    constructor (nativeAsset?: ImageSource) {
+    /**
+     * @en Constructing an ImageSource object.
+     * @zh 构造ImageSource对象
+     * @param data @en The image source. @zh 图像数据源。
+     * @deprecated @en please use `constructor(imageSource?: ImageData | IMemoryImageSource)` instead.
+     */
+    constructor (imageSource: HTMLCanvasElement | HTMLImageElement | ImageBitmap);
+    /**
+     * @en Constructing an ImageSource object.
+     * @zh 构造ImageSource对象
+     * @param data @en The image source. @zh 图像数据源。
+     */
+    constructor (imageSource?: ImageData | IMemoryImageSource);
+    constructor (imageSource?: ImageData | IMemoryImageSource | HTMLCanvasElement | HTMLImageElement | ImageBitmap) {
         super();
 
-        this._nativeData = {
-            _data: null,
-            width: 0,
-            height: 0,
-            format: 0,
-            _compressed: false,
-            mipmapLevelDataSize: [],
-        };
-
         if (EDITOR) {
             this._exportedExts = null;
         }
-
-        if (nativeAsset !== undefined) {
-            this.reset(nativeAsset);
+        if (imageSource instanceof ImageData) {
+            this._imageData = imageSource;
+        } else {
+            this._imageData = new ImageData(imageSource);
         }
     }
 
     /**
      * @en Reset the source of the image asset.
      * @zh 重置此图像资源使用的原始图像源。
      * @param data @en The new source. @zh 新的图片数据源。
+     * @deprecated @en please use `reset(data: IMemoryImageSource)` instead. @zh 请使用`reset(data: IMemoryImageSource)`替换
```

Comment:
If we deprecate DOM data type, should we add a new overload method with `ImageData` parameter?

```
public reset(data: ImageData) {...}
```

Reply:
This is not recommended, imagedata doesn't feel perfect yet, and directly recommending users to use it feels unnecessary.